src/osd/OSDMap.cc

```diff
@@ -2906,16 +2933,41 @@ class OSDTreeFormattingDumper : public CrushTreeDumper::FormattingDumper {
 public:
   typedef CrushTreeDumper::FormattingDumper Parent;
 
-  OSDTreeFormattingDumper(const CrushWrapper *crush, const OSDMap *osdmap_)
-    : Parent(crush), osdmap(osdmap_) {}
+  bool down = false;
+  bool up = false;
+  bool in = false;
+  bool out = false;
+
+  OSDTreeFormattingDumper(const CrushWrapper *crush, const OSDMap *osdmap_,
+			  const string& state)
+    : Parent(crush), osdmap(osdmap_) {
+    if (state.find("up") != std::string::npos)
```

Comment:
It's the same code as above, how about moving it to single, separate func that returns int containing flags, then passing that around?